aesara/tensor/__init__.py

```diff
@@ -45,7 +45,7 @@ def as_tensor_variable(
 @singledispatch
 def _as_tensor_variable(
     x, name: Optional[str], ndim: Optional[int], **kwargs
-) -> NoReturn:
+) -> "TensorVariable":
```

Comment:
What is the reason for this change?

Reply:
The dispatch functions are confusing mypy. It looks like it's taking the type hint of this signature and expecting it for the dispatched functions that do return the value.
Kind of like with abstract functions I think.